test/unit/specs/routes/snippet.errorTest.js

```diff
@@ -0,0 +1,177 @@
+'use strict';
+
+var chai = require('chai'),
+  sinon = require('sinon'),
+  expect = chai.expect,
+  chaiAsPromised = require('chai-as-promised'),
+  should = chai.should(),
+  Snippet = require('../../../../src/routes/snippet.js').Snippet;
+
+chai.use(chaiAsPromised);
+
+describe('Snippet upsert and delete', function() {
+  var sandbox;
+  var stubGetCorbelErrorBody;
+  var stubGetAuthorization;
+  var stubGetDriver;
+  var stubGetDomain;
+  var stubPublishAvailability;
+  var stubValidate;
+  var stubEmitEvent;
+  var stubUpsertCall;
+  var stubDeleteCall;
+
+  var req = {
+    body: {
+      id: 'test'
+    },
+    params: {
+      snippetId: 'deleteId'
+    }
+  };
+  var res = {
+    status: 200,
+    data: 'test finished'
+  };
+  var auth = {
+    auth: 'testAuth'
+  };
+  var driver = 'driver';
+  var domain = 'domain';
+  var fullId = 'FullId';
+  var textEvent = 'snippet:upsert';
+  var errorBody = 'ErrorBody';
+
+  beforeEach(function() {
+    sandbox = sinon.sandbox.create();
+
+    stubGetCorbelErrorBody = sandbox.stub(Snippet, 'getCorbelErrorBody', function(){
+        return errorBody;
+    });
+    stubGetAuthorization = sandbox.stub(Snippet, 'getAuthorization', function(){
+        return auth;
+    });
+    stubGetDriver = sandbox.stub(Snippet, 'getDriver', function(){
+        return driver;
+    });
+    stubGetDomain = sandbox.stub(Snippet, 'getDomain', function(){
+        return domain;
+    });
+    stubEmitEvent = sandbox.stub(Snippet, 'emitEvent');
+    stubDeleteCall = sandbox.stub(Snippet, 'deleteCall').returns(Promise.resolve(res));
+  });
+
+  afterEach(function() {
+    sandbox.restore();
+  });
+
+  it('Snippet.upsert throws an error in publishAvailability', function(done) {
+      var error = 'error';
+      stubPublishAvailability = sandbox.stub(Snippet, 'checkPublishAvailability', function(){
+          return Promise.reject(error);
+      });
+
+      Snippet.upsert(req, { send : function(status){
+
+        expect(status).to.be.equal(401);
+        expect(stubGetAuthorization.callCount).to.equals(1);
```

Comment:
Not sure if we need to verify all these calls or assume everything went ok implicitly... If we go this path, I'd take a look to using mocks+verify instead of stubs